docs/config.rst

```diff
@@ -751,9 +750,10 @@ Pip installer
    :default: false
    :version_added: 1.9
 
-   If ``true``, adds ``--pre`` to the ``opts`` passed to :ref:`install_command`. This will cause it to install the
+   If ``true``, adds ``PIP_PRE=1`` to the `environment <https://pip.pypa.io/en/stable/topics/configuration/#environment-variables>`_. This will cause pip it to install the
```

Comment:
This is not correct.

Setting `pip_pre = true` in a `[testenv]` _does_ cause tox to add `--pre` to the `opts`: https://github.com/tox-dev/tox/blob/997128ced47936adabb2f267d2606f8eac253ad3/src/tox/tox_env/python/pip/pip_install.py#L54-L67

The environment variable `PIP_PRE` is only set when using the _legacy_ CLI flag `--pre`:

https://github.com/tox-dev/tox/blob/4558fee5ac372c6842e5cb2ccd64c441d5a1aac5/src/tox/session/cmd/legacy.py#L60-L65

https://github.com/tox-dev/tox/blob/4558fee5ac372c6842e5cb2ccd64c441d5a1aac5/src/tox/session/cmd/legacy.py#L121-L122

Reply:
> The environment variable `PIP_PRE` is only set when using the _legacy_ CLI flag `--pre`:

Is there a plan to remove inconsistency? Or should I update the description in this PR to inform about this?

Reply:
It's my understanding that the "pip flags" accepted by the tox CLI are deprecated and will be eventually removed. The pip environment variables should be used to influence pip instead of passing (some accepted subset) of pip flags to tox.

Basically tox has no reason to maintain these pip flags now that pip exposes all of its options via environment variable.

Reply:
Ok.
But I still need to figure out what this means for this PR. Should it be closed? Modified (ex. add an explanation to avoid confusion)?

Reply:
@Czaki we have the following doc for the cli flag already: https://tox.wiki/en/latest/cli_interface.html#tox-legacy---pre

So I don't think this patch is strictly needed. However if you wanted to mention the legacy CLI flag in the testenv config section as "this config value may be overridden externally via the --pre cli flag (deprecated) or by setting PIP_PRE=1". Also note that setting PIP_PRE=0 does NOT override the testenv config, since that will use the pip cli flag, which takes precedence over envvars.

Reply:
If the user can set the pre flag, we should fix it so that it works. However, we should also document that this flag is deprecated and we encourage using the environment variables instead.

Reply:
See here